zproject/backends.py

```diff
@@ -697,6 +698,12 @@ def get_or_build_user(self, username: str, ldap_user: _LDAPUser) -> Tuple[UserPr
             # An existing user, successfully authed; return it.
             return user_profile, False
 
+        if UserProfile.objects.filter(ext_auth_uid={"ldap": ldap_user.attrs[settings.AUTH_LDAP_USERNAME_ATTR][0]}, realm=self._realm).exists():
+            user_profile = UserProfile.objects.get(ext_auth_uid={"ldap": ldap_user.attrs[settings.AUTH_LDAP_USERNAME_ATTR][0]}, realm=self._realm)
+            # Username equals mail here.
+            do_change_user_delivery_email(user_profile, username)
```

Comment:
We should think about what is the right behavior if there already is another `UserProfile` with the same email that is in `username` here. Right now I think we'll get `IntegrityError`? Should we first check whether such a user doesn't exist, and if one does, then log a warning and return without doing anything?

Reply:
I did not think about that.
I think it should be handled, because otherwise the user would see a 500

Reply:
It seems like there is no exception raised and the change of mail is just ignored.
I added a test for that. I do not know if this is the wanted behaviour